sklearn/feature_selection/tests/test_from_model.py

```diff
@@ -77,13 +105,45 @@ def test_max_features_error(max_features, err_type, err_msg):
 
 
 @pytest.mark.parametrize("max_features", [0, 2, data.shape[1]])
-def test_max_features_dim(max_features):
-    clf = RandomForestClassifier(n_estimators=50, random_state=0)
+def test_inferred_max_features_integer(max_features):
+    clf = RandomForestClassifier(n_estimators=5, random_state=0)
     transformer = SelectFromModel(
         estimator=clf, max_features=max_features, threshold=-np.inf
     )
     X_trans = transformer.fit_transform(data, y)
-    assert X_trans.shape[1] == max_features
+    # Assert correctness of inferred max features
```

Comment:
I do not think we need this comment, because the assertion code is clear in its intention. 

(Same comment about the rest of the code comments)